src/common/utils/position.ts

```diff
@@ -165,3 +168,34 @@ export const isZoomLevelGoingOutOfBounds = (value, scaleFactor: number) => {
   const b = moreThanScaleFactorMax(value, scaleFactor) && scaleFactor > 1;
   return a || b;
 };
+
+/**
+ * Get approximate Y value for given X value from a SVG line path
+ * Reference: https://stackoverflow.com/questions/11503151/in-d3-how-to-get-the-interpolated-line-data-from-a-svg-line
+ */
+export const getAprroximateYForGivenX = (
+  x: number,
+  path: SVGPathElement,
+  error: number
+) => {
+  // Tune these parameters for better accuracy
+  const MAX_BISECTION_ITERATIONS = 50;
+  error = error || 0.01;
+
+  let totalLength = path.getTotalLength();
+  let startLength = 0;
+  let middlePoint = path.getPointAtLength((totalLength + startLength) / 2);
+  let numBisectionIterations = 0;
+
+  // Binary Search
+  while (x < middlePoint.x - error || x > middlePoint.x + error) {
+    middlePoint = path.getPointAtLength((totalLength + startLength) / 2);
+    if (x < middlePoint.x) {
+      totalLength = (startLength + totalLength) / 2;
+    } else {
+      startLength = (startLength + totalLength) / 2;
+    }
+    if (MAX_BISECTION_ITERATIONS < ++numBisectionIterations) break;
```

Comment:
Always add `{ }` that way its easier to read.

```suggestion
    if (MAX_BISECTION_ITERATIONS < ++numBisectionIterations) {
      break;
    }
```